Add Support for VisualStudio 2026

```dart
CMake Error at CMakeLists.txt:3 (project):
  Generator

    Visual Studio 16 2019

  could not find any instance of Visual Studio.



Building Windows application...                                     69ms
Unable to generate build files
```

```dart
[√] Flutter (Channel stable, 3.35.5, on Microsoft Windows [Version 10.0.26100.4349], locale en-US) [507ms]
    • Flutter version 3.35.5 on channel stable at C:\SDKS\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ac4e799d23 (6 days ago), 2025-09-26 12:05:09 -0700
    • Engine revision d3d45dcf25
    • Dart version 3.9.2
    • DevTools version 2.48.0
    • Feature flags: enable-web, enable-linux-desktop, enable-macos-desktop, enable-windows-desktop, enable-android, enable-ios, cli-animations, enable-lldb-debugging

[√] Windows Version (Windows 11 or higher, 24H2, 2009) [6.4s]

[X] Android toolchain - develop for Android devices [386ms]
    X Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/to/windows-android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.


[X] Chrome - develop for the web (Cannot find Chrome executable at .\Google\Chrome\Application\chrome.exe) [340ms]
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[√] Visual Studio - develop Windows apps (Visual Studio Community 2026 18.0.0 Insiders [11018.127]) [336ms]
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\18\Insiders
    • Visual Studio Community 2026 version 18.0.11018.127
    • The current Visual Studio installation is a pre-release version. It may not be supported by Flutter yet.
    • Windows 10 SDK version 10.0.26100.0

[!] Android Studio (not installed) [176ms]
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/to/windows-android-setup for detailed instructions).

[√] IntelliJ IDEA Ultimate Edition (version 2025.2) [176ms]
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA 2025.2.2
    • Flutter plugin version 87.1
    • Dart plugin version 252.25557.23

[√] Connected device (2 available) [162ms]
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.26100.4349]
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 140.0.3485.94

[√] Network resources [1,017ms]
    • All expected network resources are available.

! Doctor found issues in 3 categories.
```

Hi, last month Microsoft released "Visual Studio 2026 Insiders" and flutter can not find it during the build process, but doctor -v sees it

--- a/packages/flutter_tools/lib/src/windows/visual_studio.dart
+++ b/packages/flutter_tools/lib/src/windows/visual_studio.dart
@@ -182,6 +182,7 @@ class VisualStudio {
   String? get cmakeGenerator {
     // From https://cmake.org/cmake/help/v3.22/manual/cmake-generators.7.html#visual-studio-generators
     return switch (_majorVersion) {
+      18 => 'Visual Studio 18 2026',
       17 => 'Visual Studio 17 2022',
       _ => 'Visual Studio 16 2019',
     };

--- a/packages/flutter_tools/test/general.shard/windows/visual_studio_test.dart
+++ b/packages/flutter_tools/test/general.shard/windows/visual_studio_test.dart
@@ -53,6 +53,18 @@ const _vs2022Response = <String, dynamic>{
   'catalog': <String, dynamic>{'productDisplayVersion': '17.0.0'},
 };
 
+// A minimum version of a response where a VS 2026 installation was found.
+const _vs2026Response = <String, dynamic>{
+  'installationPath': visualStudioPath,
+  'displayName': 'Visual Studio Community 2026',
+  'installationVersion': '18.0.11116.177',
+  'isRebootRequired': false,
+  'isComplete': true,
+  'isLaunchable': true,
+  'isPrerelease': false,
+  'catalog': <String, dynamic>{'productDisplayVersion': 'Insiders [11116.177]'},
+};
+
 // A minimum version of a response where a Build Tools installation was found.
 const _defaultBuildToolsResponse = <String, dynamic>{
   'installationPath': visualStudioPath,
@@ -823,6 +835,27 @@ void main() {
       expect(visualStudio.hasNecessaryComponents, true);
       expect(visualStudio.cmakePath, equals(cmakePath));
       expect(visualStudio.cmakeGenerator, equals('Visual Studio 17 2022'));
+      expect(visualStudio.clPath, equals(clPath));
+      expect(visualStudio.libPath, equals(libPath));
+      expect(visualStudio.linkPath, equals(linkPath));
+      expect(visualStudio.vcvarsPath, equals(vcvarsPath));
+    });
+
+    testWithoutContext('properties return the right value for Visual Studio 2026', () {
+      final VisualStudioFixture fixture = setUpVisualStudio();
+      final VisualStudio visualStudio = fixture.visualStudio;
+
+      setMockCompatibleVisualStudioInstallation(
+        _vs2026Response,
+        fixture.fileSystem,
+        fixture.processManager,
+      );
+
+      expect(visualStudio.isInstalled, true);
+      expect(visualStudio.isAtLeastMinimumVersion, true);
+      expect(visualStudio.hasNecessaryComponents, true);
+      expect(visualStudio.cmakePath, equals(cmakePath));
+      expect(visualStudio.cmakeGenerator, equals('Visual Studio 18 2026'));
       expect(visualStudio.clPath, equals(clPath));
       expect(visualStudio.libPath, equals(libPath));
       expect(visualStudio.linkPath, equals(linkPath));

